session.go

```diff
@@ -323,7 +335,12 @@ runLoop:
 }
 
 func (s *session) maybeResetTimer() {
-	deadline := s.lastNetworkActivityTime.Add(s.idleTimeout())
+	var deadline time.Time
+	if s.keepalivePingSent {
```

Comment:
This should be `if !s.keepalivePingSent && s.config.Keepalive {`

Note the `!`.